### PR TITLE
LPS-75357 Add initial drag and drop message when form builder layout is empty

### PR DESCRIPTION
**Pull request by clovisdasilvaneto**

### CODE REVIEW

**Comment by rafaprax** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder/form_builder_dd_support.js [hunk 1]
```diff
@@ -0,0 +1,730 @@
...
+			_adjustEmptyForm: function(activeLayout) {
+				var instance = this;
+
+				if (activeLayout.get('rows').length == 1 && instance._verifyEmptyForm(activeLayout.get('rows')[0]) && !A.one('#lfr-initial-col-message')) {
+					var columnMessageNode = A.Node.create('<div/>');
+
+					columnMessageNode.text(Liferay.Language.get('drag-from-sidebar-and-drop-here'));
+					columnMessageNode.setAttribute('id', 'lfr-initial-col-message');
+
+					activeLayout.get('rows')[0].get('cols')[0].get('node').addClass('lfr-initial-col');
```
Comment (rafaprax) [inline modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder/form_builder_dd_support.js]: reuse this statement `activeLayout.get('rows')[0].get('cols')[0].get('node')`  and avoid chaining

File: modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder/form_builder.js [hunk 2]
```diff
@@ -491,6 +567,95 @@ AUI.add(
...
+
+					resizeColumns: function(dragNode) {
+						var instance = this;
+
+						var dropNode = instance._layoutBuilder._lastDropEnter;
+						var lastLayoutPosition = dragNode.getData('last-layout-position');
+
+						var colLayoutPosition = dropNode.getData('layout-position');
+
+						if (!lastLayoutPosition || lastLayoutPosition != colLayoutPosition) {
```
[the remark this answers is not in the document]
Comment (clovisdasilvaneto) [reply modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/form_builder/form_builder.js]: Good, I'll do that

**Comment by vicnate5** [comment]
@rodrigo-e-c-souza We are going to remove the "Poshi Toggle" feature because of the subrepo change. I am reverting the tests back to the non-drag and drop version under LRQA-37799. The fix is actually just 1 line change:
```
<execute function="DragAndDrop" locator1="Form#ADD_FIELD_TYPE" locator2="Form#ADD_FIELD_POSITION" value1="${fieldType}" />
```
Can you make sure we get this update in once this pull is ready for merge?

@rafaprax We should remove https://github.com/rafaprax/liferay-portal/pull/257/commits/4d702f26e5c8d1a605a276bb0175e0c17fa740b9.

**Comment by rafaprax** [comment]
ci:retest

**Comment by rafaprax** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by rafaprax** [comment]
Pull request submitted to https://github.com/rodrigo-e-c-souza/liferay-portal/pull/7. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**